scripts/main/contextMenu.js

```diff
@@ -401,41 +456,42 @@ contextMenu.photoTitle = function (albumID, photoID, e) {
 	basicContext.show(items, e.originalEvent, contextMenu.close);
 };
 
+/**
+ * @param {string} photoID
+ * @param {jQuery.Event} e
+ */
 contextMenu.photoMore = function (photoID, e) {
 	// Show download-item when
 	// a) We are allowed to upload to the album
 	// b) the photo is explicitly marked as downloadable (v4-only)
 	// c) or, the album is explicitly marked as downloadable
-	let showDownload =
-		album.isUploadable() ||
-		(photo.json.hasOwnProperty("is_downloadable")
-			? photo.json.is_downloadable
-			: album.json && album.json.is_downloadable && album.json.is_downloadable);
-	let showFull = photo.json.size_variants.original.url && photo.json.size_variants.original.url !== "";
 
-	let items = [
-		{ title: build.iconic("fullscreen-enter") + lychee.locale["FULL_PHOTO"], visible: !!showFull, fn: () => window.open(photo.getDirectLink()) },
-		{ title: build.iconic("cloud-download") + lychee.locale["DOWNLOAD"], visible: !!showDownload, fn: () => photo.getArchive([photoID]) },
+	const showDownload = album.isUploadable() || photo.json.is_downloadable || (album.json && album.json.is_downloadable);
```

Comment:
```suggestion
	const showDownload = album.isUploadable() || photo.json.is_downloadable;
```

This was legacy code that was to be used only if `photo.json.is_downloadable` is not available.

Reply:
I cannot accept this suggestion anymore as it refers to outdated code. It seems this has already been fixed by another commit. So I am resolving this conversation.